### PR TITLE
BE: tests for categorical value-counts

### PR DESCRIPTION
**Pull request by ikondrat**
## Summary

Stacked on p1. Test-only PR for the value-counts implementation.

- **`tests/database/test_db_json.py`**: unit tests for `get_json_column_value_counts` — covers top-N truncation, `__other__` bucket, `__missing__` bucket, boolean fields, and empty result sets
- **`tests/api/routes/api/test_metadata.py`**: API-level integration tests for `GET /metadata/value_counts`
- **`tests/metadata/test_get_metadata_value_counts.py`**: resolver-level tests covering field selection, bucket ordering, and edge cases (all-missing field, single-value field, etc.)

## Test plan

- [ ] `make static-checks` passes
- [ ] `make test-ci` passes — all new tests green

Part of LIG-9585.

## Summary by CodeRabbit

* **New Features**
  * Added support for selecting specific categorical metadata fields when retrieving value counts.
  * Improved `/metadata/value-counts` responses and OpenAPI schema model visibility.
* **Bug Fixes**
  * Value-count requests now correctly support an omitted request body, returning empty results when no filters are provided.
* **Tests**
  * Added coverage for value-counts response shape, optional-body behavior, and OpenAPI schema models.
  * Expanded JSON extraction test coverage across DuckDB and PostgreSQL, plus SQLite unsupported-dialect handling.

### CODE REVIEW

File: lightly_studio/tests/api/routes/api/test_metadata.py [hunk 1]
```diff
@@ -75,6 +80,70 @@ def test_get_metadata_info__empty_response(test_client: TestClient, mocker: Mock
...
+
+    assert response.status_code == HTTP_STATUS_OK
+    assert response.json() == {
+        "city": {
+            "value_counts": [{"value": "Zurich", "count": 2}],
+            "other_count": 1,
+            "missing_count": 3,
+        }
+    }
+    called_filters = resolver.call_args.kwargs["filters"]
```
[the remark this answers is not in the document]
Comment (JonasWurst) [reply lightly_studio/tests/api/routes/api/test_metadata.py:110]: agree, please add

File: lightly_studio/tests/metadata/test_get_metadata_value_counts.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        ("", 1),
+        ("Missing", 1),
+        ("Other", 1),
+    ]
+    assert counts["city"].other_count == 0
+    assert counts["city"].missing_count == 2
+    assert [(entry.value, entry.count) for entry in counts["active"].value_counts] == [
+        (True, 3),
+        (False, 2),
+    ]
```
Comment (JonasWurst) [inline lightly_studio/tests/metadata/test_get_metadata_value_counts.py:69]: shouldn't we check that the correct ones are active?

File: lightly_studio/tests/metadata/test_get_metadata_value_counts.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    db_session: Session,
+) -> None:
+    collection = create_collection(session=db_session)
+    for index in range(21):
+        _create_sample(db_session, collection.collection_id, {"category": f"value-{index:02d}"})
+    _create_sample(db_session, collection.collection_id, {"category": "value-20"})
+
+    other_collection = create_collection(session=db_session)
+    for _ in range(3):
+        _create_sample(db_session, other_collection.collection_id, {"category": "value-00"})
```
Comment (JonasWurst) [inline lightly_studio/tests/metadata/test_get_metadata_value_counts.py]: why do we need 3 of them?

File: lightly_studio/tests/metadata/test_get_metadata_value_counts.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    filters = ImageFilter(
+        sample_filter=SampleFilter(
+            metadata_filters=[
+                MetadataFilter(key="city", op="==", value="A"),
+                MetadataFilter(key="group", op="==", value="x"),
+            ]
+        )
+    )
+
+    counts = categorical_value_counts.get_metadata_value_counts(
```
Comment (JonasWurst) [inline lightly_studio/tests/metadata/test_get_metadata_value_counts.py:141]: General question: are we always getting all counts for all metadata fields? Have you test this for a larger dataset? This seems very inefficient

File: lightly_studio/tests/metadata/test_get_metadata_value_counts.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    db_session: Session,
+) -> None:
+    collection = create_collection(session=db_session)
+    _create_sample(db_session, collection.collection_id, {"city": "A", "group": "x"})
+    _create_sample(db_session, collection.collection_id, {"city": "B", "group": "x"})
+    _create_sample(db_session, collection.collection_id, {"city": "B", "group": "y"})
+    filters = ImageFilter(
+        sample_filter=SampleFilter(
+            metadata_filters=[
+                MetadataFilter(key="city", op="==", value="A"),
```
Comment (JonasWurst) [inline lightly_studio/tests/metadata/test_get_metadata_value_counts.py:135]: are metadata Filters OR combined? we and combine all other filters!

File: lightly_studio/tests/metadata/test_get_metadata_value_counts.py [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+def test_get_metadata_value_counts__known_fields_with_no_matches(
+    db_session: Session,
+) -> None:
+    collection = create_collection(session=db_session)
+    _create_sample(db_session, collection.collection_id, {"city": "A"})
+
+    counts = categorical_value_counts.get_metadata_value_counts(
+        session=db_session,
+        collection_id=collection.collection_id,
+        filters=ImageFilter(width=FilterDimensions(min=10_000)),
```
Comment (JonasWurst) [inline lightly_studio/tests/metadata/test_get_metadata_value_counts.py:193]: why doe this hit?

File: lightly_studio/tests/metadata/test_get_metadata_value_counts.py [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+
+
+def test_get_metadata_value_counts__unknown_collection_is_empty(db_session: Session) -> None:
+    counts = categorical_value_counts.get_metadata_value_counts(
+        session=db_session, collection_id=uuid4()
+    )
+    assert counts == {}
+
+
+def test_get_metadata_value_counts__literal_top_level_keys(db_session: Session) -> None:
```
Comment (JonasWurst) [inline lightly_studio/tests/metadata/test_get_metadata_value_counts.py:206]: I do not understand the purpose of this test

**Review by JonasWurst — CHANGES_REQUESTED** [review]

File: lightly_studio/src/lightly_studio/api/routes/api/metadata.py [hunk 8]
```diff
@@ -96,13 +96,20 @@ class MetadataValueCountsRequest(BaseModel):
     """Request body for computing filtered categorical value counts."""
 
     filters: ImageFilter | None = Field(None, description="Filter parameters for samples")
+    fields: list[str] | None = Field(
```
Comment (ikondrat) [inline lightly_studio/src/lightly_studio/api/routes/api/metadata.py:99]: missed update for request details

**Review by JonasWurst — APPROVED** [review]
